### ISSUE TITLE
--continue_from checkpoint results in invalid types error  (with --cuda)

### ISSUE THREAD

**Issue by mschonwe** [SeanNaren/deepspeech.pytorch#195]
Command line:
```
python train.py --train_manifest data/an4_train_manifest.csv --val_manifest data/an4_val_manifest.csv --save_folder "an4_models/" --model_path "an4_models/ds_final.pth.tar" --cuda --checkpoint --checkpoint_per_batch 20 --continue_from an4_models/deepspeech_checkpoint_epoch_2_iter_40.pth.tar
```

Error Message:
```
Traceback (most recent call last):
  File "train.py", line 264, in <module>
    optimizer.step()
  File "/usr/local/lib/python2.7/dist-packages/torch/optim/sgd.py", line 94, in step
    buf.mul_(momentum).add_(1. - dampening, d_p)
TypeError: add_ received an invalid combination of arguments - got (float, torch.cuda.FloatTensor), but expected one of:
 * (float value)
 * (torch.FloatTensor other)
 * (torch.SparseFloatTensor other)
 * (float value, torch.FloatTensor other)
      didn't match because some of the arguments have invalid types: (float, torch.cuda.FloatTensor)
 * (float value, torch.SparseFloatTensor other)
      didn't match because some of the arguments have invalid types: (float, torch.cuda.FloatTensor)
```

It will resume from the checkpoint without the --cuda flag.

Any suggestions?

EDIT:
Not sure if this matters, but I'm using cuda 8 and cudnn 6.

**Comment by dalcazar** [comment]
Same for me.

I've tried with CUDA 8 and a lot of pytorch and python version (2.7 / 3.5) combinations and the same error always. It happens to me since I've updated to last version one month ago.

**Comment by ryanleary** [comment]
Yeah this is a bug in the order the optimizer is initialized I think. Open to PRs, it should be a simple reordering. I won't be able to address it until next week at the earliest.

**Comment by AMairesse** [comment]
Duplicate of #210

**Comment by ryanleary** [comment]
Closed in #217.
